### ISSUE TITLE
Outbox message never leave Outbox, even with calling manually "Send messages"

### ISSUE THREAD

**Issue by liar666** [thundernest/k-9#5658]
Hi,

As I reported here:
https://forum.k9mail.app/t/emails-in-outbox-before-upgrade-cannot-be-sent-after-upgrade/2917/2
I've upgraded K9 to the beautiful new version while having some messages in the outbox.

Now, even if I hit "Send Messages" inside the Outbox dir, the messages are not sent. There's a '!' shown for a few milliseconds in the android notifications bar, but it's so fast I cannot read it.

I've also might have done an android "cache cleaning" for K9 app in the meantime, but I still can access the messages and their contents in the Outbox, so this should not be the problem...

New message are sent without any problem (even if I'm not online, they're sent when I hit "Send Messages" in the Outbox dir).

I've rebooted (to try to removed any hard "reseng" limit), but this does not solve the problem.

I've activated logs, here is what I get:
```
--------- beginning of main
09-10 16:10:14.944  9247  9275 I OpenGLRenderer: queryNativeWindowStatus 0x6f6c632010 query=0 dataspace=0
09-10 16:10:14.966  9247  9275 D         : GBuffer free  w:1080, h:2280, f:0x111, usage:0xb00, line = 1909
09-10 16:10:14.967  9247  9275 D         : GBuffer alloc ----------------- line= 2565,GrallocGetBufferFormat=NULL
09-10 16:10:14.969  9247  9275 I chatty  : uid=10117(com.fsck.k9) RenderThread identical 6 lines
09-10 16:10:14.971  9247  9275 D         : GBuffer alloc ----------------- line= 2565,GrallocGetBufferFormat=NULL
09-10 16:10:15.037  9247  9275 D         : GBuffer free  w:1080, h:2280, f:0x111, usage:0xb00, line = 1909
09-10 16:10:15.055  9247  9275 D         : GBuffer free  w:1080, h:2280, f:0x111, usage:0xb00, line = 1909
09-10 16:10:16.272  9247  9275 D OpenGLRenderer: textureCacheSize 75497472
09-10 16:10:16.312  9247  9275 I OpenGLRenderer: queryNativeWindowStatus 0x6f6c9cb010 query=0 dataspace=0
09-10 16:10:16.327  9247  9275 D OpenGLRenderer: textureCacheSize 75497472
09-10 16:10:16.337  9247  9275 D         : GBuffer free  w:780, h:624, f:0x111, usage:0xb00, line = 1909
09-10 16:10:16.337  9247  9275 D         : GBuffer alloc ----------------- line= 2565,GrallocGetBufferFormat=NULL
09-10 16:10:16.340  9247  9275 I chatty  : uid=10117(com.fsck.k9) RenderThread identical 6 lines
09-10 16:10:16.341  9247  9275 D         : GBuffer alloc ----------------- line= 2565,GrallocGetBufferFormat=NULL
09-10 16:10:16.352  9247  9275 D         : GBuffer free  w:780, h:624, f:0x111, usage:0xb00, line = 1909
09-10 16:10:16.368  9247  9275 D         : GBuffer free  w:780, h:624, f:0x111, usage:0xb00, line = 1909
09-10 16:10:17.229  9247  9262 I MessagingController: Running command 'sendPendingMessages', seq = 5 (background priority)
09-10 16:10:17.287  9247  9262 I MessagingController: Scanning Outbox folder for messages to send
09-10 16:10:17.288  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e67680f1-2180-4137-90f4-02f9afcc1d5d
09-10 16:10:17.294  9247  9262 V MessagingController: Skipping sending message K9LOCAL:35c78b58-650d-405f-9f27-ca57de9bd198
09-10 16:10:17.300  9247  9262 V MessagingController: Skipping sending message K9LOCAL:29c95f84-01ea-4a91-88b6-f840fd0ef4c2
09-10 16:10:17.305  9247  9262 V MessagingController: Skipping sending message K9LOCAL:24bb3a2f-9388-4577-90c8-8dfe3abe595b
09-10 16:10:17.311  9247  9262 V MessagingController: Skipping sending message K9LOCAL:28a36290-c17f-4c82-b11e-eb10a6b74ce2
09-10 16:10:17.317  9247  9262 V MessagingController: Skipping sending message K9LOCAL:5d3a12f2-88a3-41ba-ad47-439e7b843650
09-10 16:10:17.323  9247  9262 V MessagingController: Skipping sending message K9LOCAL:0fda83c8-3df6-45b3-99c3-abab436495be
09-10 16:10:17.329  9247  9262 V MessagingController: Skipping sending message K9LOCAL:b6bec1bb-557b-4f4f-8b02-3c2360ba9c4a
09-10 16:10:17.334  9247  9262 V MessagingController: Skipping sending message K9LOCAL:3c99910a-f4b8-4bd9-8a61-caa1b5f7179f
09-10 16:10:17.342  9247  9262 V MessagingController: Skipping sending message K9LOCAL:04cd8c11-6dc4-4e99-b256-794a8500ae43
09-10 16:10:17.347  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e4b9eeb8-c28e-4756-8637-1a7e623fd226
09-10 16:10:17.351  9247  9262 V MessagingController: Skipping sending message K9LOCAL:2b66f46c-2095-4eaf-a317-ba82dde03153
09-10 16:10:17.356  9247  9262 V MessagingController: Skipping sending message K9LOCAL:0dd143db-5f1d-40dc-b20c-00173eeee33a
09-10 16:10:17.361  9247  9262 V MessagingController: Skipping sending message K9LOCAL:96811236-bbc9-4904-94ab-9f57dc1ebd52
09-10 16:10:17.367  9247  9262 V MessagingController: Skipping sending message K9LOCAL:4101e255-6a9d-4298-b002-7333c90b5ac5
09-10 16:10:17.371  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e38eebeb-7824-4c1a-8fea-f448fdcd439b
09-10 16:10:17.376  9247  9262 V MessagingController: Skipping sending message K9LOCAL:847b2f06-6f9e-44e6-9fea-14870b0138fe
09-10 16:10:17.382  9247  9262 V MessagingController: Skipping sending message K9LOCAL:a31d64ec-42d0-4bd0-8199-05ab04ec752f
09-10 16:10:17.386  9247  9262 V MessagingController: Skipping sending message K9LOCAL:dc9da175-4cc8-4c5e-acf5-135ca3de8649
09-10 16:10:17.391  9247  9262 V MessagingController: Skipping sending message K9LOCAL:300dfb67-a3fc-413f-afbf-e66a6ca79913
09-10 16:10:17.398  9247  9262 V MessagingController: Skipping sending message K9LOCAL:a0758d1f-6477-454b-9521-994c2cb9ce8b
09-10 16:10:17.403  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e4332321-48c1-4538-801c-4491b3eec59b
09-10 16:10:17.409  9247  9262 V MessagingController: Skipping sending message K9LOCAL:8545a445-983f-4a0b-85f5-b40cb262db75
09-10 16:10:17.414  9247  9262 V MessagingController: Skipping sending message K9LOCAL:2fefc2e1-143e-4380-87f4-78a57c87b18a
09-10 16:10:17.419  9247  9262 V MessagingController: Skipping sending message K9LOCAL:206224a9-d037-4cee-a654-32637af23ec5
09-10 16:10:17.425  9247  9262 V MessagingController: Skipping sending message K9LOCAL:cad861dd-0fec-4e87-a819-846db160ca61
09-10 16:10:17.430  9247  9262 V MessagingController: Skipping sending message K9LOCAL:7099de21-8692-4e3e-ab74-9930fda9c16e
09-10 16:10:17.435  9247  9262 V MessagingController: Skipping sending message K9LOCAL:725f0182-62a6-401e-957e-38b32b718336
09-10 16:10:17.441  9247  9262 V MessagingController: Skipping sending message K9LOCAL:8b185cea-7e31-4519-a3b3-b4bac0e3c199
09-10 16:10:17.446  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e574d340-9e82-4426-b87b-e6b23bca12b9
09-10 16:10:17.451  9247  9262 V MessagingController: Skipping sending message K9LOCAL:5f9b83e4-3562-4f19-8d7e-1bf026809f8f
09-10 16:10:17.456  9247  9262 V MessagingController: Skipping sending message K9LOCAL:23a2bd3f-96af-4985-a8b1-ce8cf56b934e
09-10 16:10:17.461  9247  9262 V MessagingController: Skipping sending message K9LOCAL:3d7e0c13-d7c7-40d8-8aed-8fe2a3a7d3da
09-10 16:10:17.467  9247  9262 V MessagingController: Skipping sending message K9LOCAL:5cb6a04e-87f7-4726-9577-69adf8e231fc
09-10 16:10:17.471  9247  9262 V MessagingController: Skipping sending message K9LOCAL:bcd98710-da41-4c38-8d55-406d3c5ed633
09-10 16:10:17.479  9247  9262 V MessagingController: Skipping sending message K9LOCAL:209ffa0d-eeba-4a5d-8f78-1853a2cca430
09-10 16:10:17.484  9247  9262 V MessagingController: Skipping sending message K9LOCAL:6a9eb352-cd08-41bf-891a-62cd54efa7c3
09-10 16:10:17.489  9247  9262 V MessagingController: Skipping sending message K9LOCAL:66a76e1b-56c1-41e5-b52e-146f8d9b7e6e
09-10 16:10:17.495  9247  9262 V MessagingController: Skipping sending message K9LOCAL:1849a72f-a8ee-40cf-af3f-e8a1c9d0c5a5
09-10 16:10:17.501  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e9ed630b-ed4d-4e6e-b71f-a3c124f13ef6
09-10 16:10:17.506  9247  9262 V MessagingController: Skipping sending message K9LOCAL:fceb00e3-169e-407c-8b1f-99dc1aa10da2
09-10 16:10:17.511  9247  9262 V MessagingController: Skipping sending message K9LOCAL:45971d09-83aa-4b05-b601-9fb18cb79b78
09-10 16:10:17.517  9247  9262 V MessagingController: Skipping sending message K9LOCAL:3462b924-a834-4772-ba36-bb029a399068
09-10 16:10:17.522  9247  9262 V MessagingController: Skipping sending message K9LOCAL:cf320e32-1b29-4e14-aa21-b1ac85198d96
09-10 16:10:17.527  9247  9262 V MessagingController: Skipping sending message K9LOCAL:1fd9a5e6-4737-41b1-802e-a2e11d13984f
09-10 16:10:17.534  9247  9262 V MessagingController: Skipping sending message K9LOCAL:0f8255f8-0f6a-4cc2-9cd6-34d081aa3344
09-10 16:10:17.540  9247  9262 V MessagingController: Skipping sending message K9LOCAL:1ba4d9f3-c5e6-4e64-8a34-3808ab27977f
09-10 16:10:17.546  9247  9262 V MessagingController: Skipping sending message K9LOCAL:06e85e63-91a4-40df-9713-c37496caea71
09-10 16:10:17.551  9247  9262 V MessagingController: Skipping sending message K9LOCAL:764e7fe0-aec5-4f5b-ad5c-2109d84d427c
09-10 16:10:17.556  9247  9262 V MessagingController: Skipping sending message K9LOCAL:8a0b3f3a-8607-4619-8cbc-c1993c1b8aec
09-10 16:10:17.561  9247  9262 V MessagingController: Skipping sending message K9LOCAL:1dc42dec-7dfb-450e-a2fc-b8020eb6446f
09-10 16:10:17.567  9247  9262 V MessagingController: Skipping sending message K9LOCAL:71486bab-113c-4bb5-852b-a906b69e675f
09-10 16:10:17.574  9247  9262 V MessagingController: Skipping sending message K9LOCAL:049c1f1b-4fd3-4cc5-8fbc-6f5ee05b88b9
09-10 16:10:17.576  9247  9275 D OpenGLRenderer: endAllActiveAnimators on 0x6f7c9ff800 (MenuPopupWindow$MenuDropDownListView) with handle 0x6f6f4757a0
09-10 16:10:17.581  9247  9262 V MessagingController: Skipping sending message K9LOCAL:65df4c73-7c2a-4824-b4a3-00290c032faf
09-10 16:10:17.587  9247  9262 V MessagingController: Skipping sending message K9LOCAL:29fb680a-06cf-4db3-88a8-465b8729fe47
09-10 16:10:17.594  9247  9262 V MessagingController: Skipping sending message K9LOCAL:beb61c2a-5d12-4872-a28b-18f40143633a
09-10 16:10:17.600  9247  9262 V MessagingController: Skipping sending message K9LOCAL:91c048b6-a878-4e3f-93f6-77a31dcf9a1c
09-10 16:10:17.605  9247  9262 V MessagingController: Skipping sending message K9LOCAL:094e5624-8ac5-4dae-9f3c-194373b52ea5
09-10 16:10:17.611  9247  9262 V MessagingController: Skipping sending message K9LOCAL:b33c00ac-ee1c-42e2-8385-72d95dcc227f
09-10 16:10:17.616  9247  9262 V MessagingController: Skipping sending message K9LOCAL:72ba1df4-7c87-4339-a54a-2e43abebd530
09-10 16:10:17.622  9247  9262 V MessagingController: Skipping sending message K9LOCAL:44ceea8f-899c-48e7-8bec-702123f91b18
09-10 16:10:17.627  9247  9262 V MessagingController: Skipping sending message K9LOCAL:2c4d2e67-45df-41b2-9c93-614d985951ef
09-10 16:10:17.632  9247  9262 V MessagingController: Skipping sending message K9LOCAL:37129251-99b2-44b1-b210-8d26e8bac543
09-10 16:10:17.638  9247  9262 V MessagingController: Skipping sending message K9LOCAL:0a629150-188a-467b-b521-a9c35bcb2af9
09-10 16:10:17.642  9247  9262 V MessagingController: Skipping sending message K9LOCAL:7c045a08-5059-4ec3-b1cb-0f7b89ff0487
09-10 16:10:17.647  9247  9262 V MessagingController: Skipping sending message K9LOCAL:eb47fd51-6c3f-41fa-913c-110a4aec531d
09-10 16:10:17.653  9247  9262 V MessagingController: Skipping sending message K9LOCAL:88074487-2b97-4aa3-bf52-1c448a774b11
09-10 16:10:17.658  9247  9262 V MessagingController: Skipping sending message K9LOCAL:6108c858-739d-4567-89a5-50356d5dc9f9
09-10 16:10:17.664  9247  9262 V MessagingController: Skipping sending message K9LOCAL:6d42aae0-c8cc-49d6-b72e-0091b4b7e1dc
09-10 16:10:17.669  9247  9262 V MessagingController: Skipping sending message K9LOCAL:27ff8910-6e92-4aaf-8e5b-ee1272e2a67a
09-10 16:10:17.673  9247  9262 V MessagingController: Skipping sending message K9LOCAL:79cb42ac-c16d-41c3-a29b-ae982d5a3d17
09-10 16:10:17.679  9247  9262 V MessagingController: Skipping sending message K9LOCAL:52aba970-405c-4f2d-a156-ca7790861f4b
09-10 16:10:17.684  9247  9262 V MessagingController: Skipping sending message K9LOCAL:540f0c46-d04b-4edd-afbd-10f935199e5d
09-10 16:10:17.688  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e665300a-2d1c-457c-b0e3-e8a67a24c076
09-10 16:10:17.694  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e961eec4-17ac-4a18-9b4d-a1af8d4d4cb4
09-10 16:10:17.698  9247  9262 V MessagingController: Skipping sending message K9LOCAL:661cc538-56d9-49d2-9490-679c4c0464ce
09-10 16:10:17.703  9247  9262 V MessagingController: Skipping sending message K9LOCAL:4c47115f-4052-4894-8150-3cb6f81fcc49
09-10 16:10:17.707  9247  9262 V MessagingController: Skipping sending message K9LOCAL:848bdd03-8597-40fe-946d-1effac647be7
09-10 16:10:17.713  9247  9262 V MessagingController: Skipping sending message K9LOCAL:3c2970ac-b4e4-4b5e-83f5-2919bf4bdc94
09-10 16:10:17.718  9247  9262 V MessagingController: Skipping sending message K9LOCAL:74a7d7ea-b417-44e4-9ab9-ce8bb7a247ee
09-10 16:10:17.724  9247  9262 V MessagingController: Skipping sending message K9LOCAL:0ba4774c-903c-4df6-9471-56be400e9917
09-10 16:10:17.730  9247  9262 V MessagingController: Skipping sending message K9LOCAL:9ed17418-3ddc-4da3-810b-2c0dc7e89794
09-10 16:10:17.735  9247  9262 V MessagingController: Skipping sending message K9LOCAL:09df6447-73a2-42d1-8245-44dab9ca6750
09-10 16:10:17.741  9247  9262 V MessagingController: Skipping sending message K9LOCAL:5c0d08db-ac7a-4fa4-9f60-bf1584b8d071
09-10 16:10:17.748  9247  9262 V MessagingController: Skipping sending message K9LOCAL:f53e9da0-631a-47e3-8019-23d38dea4b17
09-10 16:10:17.753  9247  9262 V MessagingController: Skipping sending message K9LOCAL:d9bca822-aa43-4d55-b7c3-60828ddc27e7
09-10 16:10:17.758  9247  9262 V MessagingController: Skipping sending message K9LOCAL:df11d16f-7c83-4632-9c01-520ddaa864dd
09-10 16:10:17.764  9247  9262 V MessagingController: Skipping sending message K9LOCAL:be05a3ea-2095-4b97-9d1e-0faf36a737a5
09-10 16:10:17.770  9247  9262 V MessagingController: Skipping sending message K9LOCAL:14776c89-078d-4060-aea2-398f35edc92b
09-10 16:10:17.776  9247  9262 V MessagingController: Skipping sending message K9LOCAL:0fe9c34d-b3b3-44db-87e9-cf98d248c4ad
09-10 16:10:17.782  9247  9262 V MessagingController: Skipping sending message K9LOCAL:4593565b-b2e7-4b7b-8e2e-860ad74b37c4
09-10 16:10:17.788  9247  9262 V MessagingController: Skipping sending message K9LOCAL:d49ed4bb-5d8b-49d0-a69d-18804adb41ce
09-10 16:10:17.794  9247  9262 V MessagingController: Skipping sending message K9LOCAL:5ac1c8aa-944e-4c9d-ba91-653b51ba066d
09-10 16:10:17.799  9247  9262 V MessagingController: Skipping sending message K9LOCAL:7a3e0399-098b-4383-8278-0e013f57b70c
09-10 16:10:17.805  9247  9262 V MessagingController: Skipping sending message K9LOCAL:049db3ac-7ced-4f5e-a9bc-535d509c6f96
09-10 16:10:17.811  9247  9262 V MessagingController: Skipping sending message K9LOCAL:df7b7425-8e1a-419c-a6c8-d398f9d7e5d9
09-10 16:10:17.816  9247  9262 V MessagingController: Skipping sending message K9LOCAL:2ff5cb1b-2d46-48e5-8067-b83c06d5d0d5
09-10 16:10:17.822  9247  9262 V MessagingController: Skipping sending message K9LOCAL:96135bdb-4906-443d-a05b-dc0260793a5e
09-10 16:10:17.828  9247  9262 V MessagingController: Skipping sending message K9LOCAL:a6c45d3a-7e45-473e-bba5-01dfede567c0
09-10 16:10:17.833  9247  9262 V MessagingController: Skipping sending message K9LOCAL:739dbee8-06ae-4ad1-8c6e-3fbad9a17f48
09-10 16:10:17.839  9247  9262 V MessagingController: Skipping sending message K9LOCAL:7c572d69-5ceb-43bd-bf35-6f0ef51aa09b
09-10 16:10:17.844  9247  9262 V MessagingController: Skipping sending message K9LOCAL:bd648c28-6886-49f8-b820-4549a19480b5
09-10 16:10:17.850  9247  9262 V MessagingController: Skipping sending message K9LOCAL:18c95055-f268-4cd0-b61d-c9512e8924e5
09-10 16:10:17.855  9247  9262 V MessagingController: Skipping sending message K9LOCAL:3aaa45a5-c71e-4979-ae91-34dbff127ccb
09-10 16:10:17.861  9247  9262 V MessagingController: Skipping sending message K9LOCAL:be4c8b6e-0999-4daf-aa57-6c490004bb30
09-10 16:10:17.867  9247  9262 V MessagingController: Skipping sending message K9LOCAL:8ceacd53-f3a3-4419-8270-3637da80e60e
09-10 16:10:17.873  9247  9262 V MessagingController: Skipping sending message K9LOCAL:582f8779-9dfb-4651-aeb7-2e095f5b462c
09-10 16:10:17.879  9247  9262 V MessagingController: Skipping sending message K9LOCAL:ae76da2a-0fc5-4f67-932e-0b2e0a361146
09-10 16:10:17.884  9247  9262 V MessagingController: Skipping sending message K9LOCAL:f72f2142-003a-419d-9be9-ba7ceb118804
09-10 16:10:17.890  9247  9262 V MessagingController: Skipping sending message K9LOCAL:bad06a1c-c786-4b83-8384-e9e205e5cbd7
09-10 16:10:17.895  9247  9262 V MessagingController: Skipping sending message K9LOCAL:6953eacc-f235-48d0-92fb-6b8f74c02e80
09-10 16:10:17.901  9247  9262 V MessagingController: Skipping sending message K9LOCAL:63e32b80-aee9-4e24-a8ff-e703596b9a90
09-10 16:10:17.907  9247  9262 V MessagingController: Skipping sending message K9LOCAL:ecf08e5f-5959-452f-84fb-2a25b39a7561
09-10 16:10:17.912  9247  9262 V MessagingController: Skipping sending message K9LOCAL:88fa1946-245f-4e58-8c0e-03ce4c48bf44
09-10 16:10:17.918  9247  9262 V MessagingController: Skipping sending message K9LOCAL:72d5f8dd-9fdd-43ff-8e25-cb1ddb4fc18d
09-10 16:10:17.924  9247  9262 V MessagingController: Skipping sending message K9LOCAL:c342043c-0849-45e1-9fc5-6e7f1ddeabf6
09-10 16:10:17.930  9247  9262 V MessagingController: Skipping sending message K9LOCAL:91f04217-65a1-4804-88ac-18e0d19e6540
09-10 16:10:17.936  9247  9262 V MessagingController: Skipping sending message K9LOCAL:8e35576f-c220-443e-8342-d2a9c0f41b90
09-10 16:10:17.942  9247  9262 V MessagingController: Skipping sending message K9LOCAL:e99163ab-9310-4b63-baff-5b8a33be24bf
09-10 16:10:17.948  9247  9262 V MessagingController: Skipping sending message K9LOCAL:bbfcb306-4d2a-4802-935c-fb9cacced898
09-10 16:10:17.954  9247  9262 V MessagingController: Skipping sending message K9LOCAL:cf69f246-edd3-48bf-908e-33df581d2ae6
09-10 16:10:17.959  9247  9262 V MessagingController: Skipping sending message K9LOCAL:9c8e08cc-b751-445b-9439-77faab6c401b
09-10 16:10:17.965  9247  9262 V MessagingController: Skipping sending message K9LOCAL:791e7261-e987-44dc-8028-8ce30dd5a8a4
09-10 16:10:17.971  9247  9262 I MessagingController:  Command 'sendPendingMessages' completed
```

Why are the messages simply ignored ??!?! Now, I'm stuck with 123 messages in my Outbox :{

There seem to be related issues, like:
https://github.com/k9mail/k-9/issues/4594
https://github.com/k9mail/k-9/issues/613
but they seem to have been more or less solved, whereas mine is not.

Any solution?

**Comment by megfault** [comment]
I just assisted a friend with her K9 configuration, she also could not get any messages out, but incoming mail worked.

It turned out she changed passwords at some point, but - and I tested this extensively today - in the SMTP settings form, the displayed password is not the one being used.  She had to type the old password in order for the connection to work. Other applications like Thunderbird were perfectly happy with the new password, as well as the IMAP config on K9.

I know this sounds completely bonkers, but hear me out: It seems like both the current password and the previous password are being stored persistently, and something is going very wrong in the form processing logic. Hence it looked like the account data was correct but the messages just didn't "want" to leave. Perhaps this is what happening to you too.

If I have any time and energy I will try to confirm this in the source code, perhaps even fix it. Until then, I hope my account of this weird experience helps someone.

**Comment by eternaltyro** [comment]
I can confirm the same behaviour with my instance as well. I use Zoho as the email provider and all sent emails remain in the Outbox.

Version: 5.907

**Comment by hills** [comment]
I've had this problem for many years with K9. Previously I didn't use my phone much to compose messages, but now that has changed and I'd like to get closer to a fix or even workaround.

Usually when these messages fail to send the first time, it's because I am on a WiFi network but not yet signed in. The initial failure is of the sort "Hostname 'mail.example.com' is not known". Perhaps this is being treated as some kind of permanent error?

In reality, since there isn't a good quality return path for errors, perhaps all should be treated as intermittent, and re-tried at reasonable intervals.

Because I have messages quietly sitting in the Outbox for several weeks now, despite re-connecting to various Wifi and 4G networks for long periods.

If I go to "Outbox" and explicitly run "Send messages" from the menu this appears to have no effect. See logs of this action, which are similar to the above:

```
08-16 22:25:58.537 19741 19763 I MessagingController: Running command 'sendPendingMessages', seq = 2666 (background priority)
08-16 22:25:58.554 19741 19763 I MessagingController: Scanning Outbox folder for messages to send
08-16 22:25:58.557 19741 19763 V MessagingController: Skipping sending message K9LOCAL:0408762a-e196-4f96-af4b-e35a575fbaae
08-16 22:25:58.568 19741 19763 I MessagingController:  Command 'sendPendingMessages' completed
```

**Comment by hills** [comment]
That message "skipping sending message" originates [from here](https://github.com/thundernest/k-9/blob/1d10f7684025cad4a73a085ed7f43825716d3203/app/core/src/main/java/com/fsck/k9/controller/MessagingController.java#L1561)

So the message does not satisfy this little fragment

```
long messageId = message.getDatabaseId();
OutboxState outboxState = outboxStateRepository.getOutboxState(messageId);

if (outboxState.getSendState() != SendState.READY) {
```

SendState can be one of:

```
READY("ready"),
RETRIES_EXCEEDED("retries_exceeded"),
ERROR("error");
```

so there is scope for a "retries exceeded" or "error" cases. I'm not familiar with this codebase (or Kotlin code) but perhaps it would be a start to log this piece of information to know why messages that are perfectly valid are not sending.

**Comment by cketti** [comment]
Issue #4602 is a reminder to improve our handling of send failures.

PR #6283 improves the error reporting for send failures and will close this issue once merged.

**Comment by LinuxOnTheDesktop** [comment]
Has the new handling of send failures made it into a release? I find that I continue to have the problem. Indeed: [while initially the problem owed to my having entered an incorrect server address](https://forum.k9mail.app/t/emails-stuck-in-outbox/4361/25), the problem has reoccurred now, even though I have the correct server address set. There are some logs (of the current version of the problem) available via [here](https://forum.k9mail.app/t/emails-stuck-in-outbox-semi-permanently-on-k9-6-602/6692).

**Comment by codethief** [comment]
I just ran into this ("Retries exceeded"). I have been experiencing network issues today, so the email in my outbox didn't get sent successfully. At some point, it became impossible to retry – K9/Thunderbird always responded with "Retries exceeded". My workaround was to move the email back to Drafts and then send it again. Clearly, this is not great, though.

**Comment by hills** [comment]
This continues to be a problem for me. I really think the concept of RETRIES_EXCEEDED is incorrectly designed; what it is trying to protect? If a message in the outbox then delivery should be attempted at reasonable intervals until successful.

**Comment by codethief** [comment]
@hills I agree. Though I suppose this is addressed by the first item in #4602

**Comment by hills** [comment]
> [@hills](https://github.com/hills) Though I suppose this is addressed by the first item in [#4602](https://github.com/thunderbird/thunderbird-android/issues/4602).

>> Retry sending messages that have exceeded the maximum retry count when the user selects "send messages" from the Outbox menu

It might get us me of a hole (that's if it works, you seem to have found a bug and should probably "reopen issue"?)

But the reason I have a computer is for it to do useful work for me.

Write an email from deep on the London Underground and you've already exceeded the retry count by the time you come above ground. (Travel through a few stations but never in one long enough to fully make a connection.)

For the above to be effective, I need to see an alert, look in the outbox and select the menu. Which often doesn't happen until hours (or days) later.

But during all this time K9 has been delegated clear instructions that I want it to send the mail.
